.github/workflows/update-copyright-notices.yml

```diff
@@ -0,0 +1,46 @@
+name: Update copyright notices
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types:
+      - closed
+
+jobs:
+  check-and-update-copyright-notices:
+    if: ${{ github.event_name == 'push' }}
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - run: ./tools/update-copyright-notices.sh
+      - env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          git config --global user.email "actions@github.com"
+          git config --global user.name "github-actions[bot]"
+          pr_branch=updateCopyrightNotices
+          git checkout -b ${pr_branch}
+          git add .
+          git commit -m "AUTO: Update copyright notices for $(date +%Y)" || exit 0
+          git push origin "+${pr_branch}"
+          existing_pr=$(gh pr list --head "${pr_branch}" --json number --jq '.[].number')
+          if [[ $existing_pr ]]; then
+            gh pr comment "${existing_pr}" --body "PR has been updated again."
+            exit 0
```

Comment:
I'd prefer a if/else instead of exiting here (this makes the code clearer)

Reply:
Done.

Reply:
Personally, I am a big fan of early exit/return, and find it much clearer.

If there is nothing further to do, why not exit at that point, rather than have the reader need to follow through the else/fi to find there is nothing else below?

Reply:
> Personally, I am a big fan of early exit/return, and find it much clearer.

Same here. I guess this instance is less clear than others though as there are two code paths where neither is the clear default. If the script was to be extended, I couldn't predict which of those two paths it would be.